src/main/java/seedu/address/storage/XmlAdaptedRoom.java

```diff
@@ -0,0 +1,148 @@
+package seedu.address.storage;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.xml.bind.annotation.XmlElement;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.person.Guest;
+import seedu.address.model.room.Capacity;
+import seedu.address.model.room.DoubleRoom;
+import seedu.address.model.room.Expenses;
+import seedu.address.model.room.Reservations;
+import seedu.address.model.room.Room;
+import seedu.address.model.room.RoomNumber;
+import seedu.address.model.room.SingleRoom;
+import seedu.address.model.room.SuiteRoom;
+import seedu.address.model.tag.Tag;
+
+/**
+ * JAXB-friendly version of the Room.
+ */
+public class XmlAdaptedRoom {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Room's %s field is missing!";
+
+    @XmlElement(required = true)
+    private String roomNumber;
+    @XmlElement(required = true)
+    private String capacity;
+
+    @XmlElement
+    private String expenses;
+    @XmlElement
+    private String reservations;
+    @XmlElement
+    private List<Guest> guests = new ArrayList();
+    @XmlElement
+    private List<XmlAdaptedTag> tagged = new ArrayList<>();
+
+    /**
+     * Constructs an XmlAdaptedPerson.
+     * This is the no-arg constructor that is required by JAXB.
+     */
+    public XmlAdaptedRoom() {}
+
+    /**
+     * Constructs an {@code XmlAdaptedPerson} with the given guest details.
+     */
+    public XmlAdaptedRoom(String roomNumber, String capacity, String expenses,
+                          String reservations, List<Guest> guests, List<XmlAdaptedTag> tagged) {
+        this.roomNumber = roomNumber;
+        this.capacity = capacity;
+        this.expenses = expenses;
+        this.reservations = reservations;
+
+        if (guests != null) {
+            this.guests = new ArrayList<>(guests);
+        }
+        if (tagged != null) {
+            this.tagged = new ArrayList<>(tagged);
+        }
+    }
+
+    /**
+     * Converts a given Guest into this class for JAXB use.
+     *
+     * @param source future changes to this will not affect the created XmlAdaptedPerson
+     */
+    public XmlAdaptedRoom(Room source) {
+        roomNumber = source.getRoomNumber().toString();
+        capacity = source.getCapacity().toString();
+        expenses = source.getExpenses().toString();
+        reservations = source.getReservations().toString();
+        guests = source.getOccupant();
+        tagged = source.getTags().stream()
+                .map(XmlAdaptedTag::new)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Converts this jaxb-friendly adapted guest object into the model's Guest object.
```

Comment:
so many

Reply:
Sorry lah i tried to find all these at the end but i missed out a whole chunk